src/utils/common/index.ts

```diff
@@ -1 +1,33 @@
 export const noop = (): void => undefined
+
+export const find = <T extends any[] | readonly any[]>(
+  arr: T,
+  target: UnknownObject<string | number>
+): T extends readonly any[] ? ValueOf<T> : undefined => {
+  const result = arr.find(item => {
+    const splittedArr = splitObject(item)
+
+    return splittedArr.some(splittedItem => {
+      const [itemKey, itemValue] = Object.entries(splittedItem).flat()
+      const [targetKey, targetValue] = Object.entries(target).flat()
+
+      return itemKey === targetKey && itemValue === targetValue
+    })
+  })
+
+  return result
+}
+
+type SplittedObjectArray<T> = { [key in KeyOf<T>]: ValueOf<T> }[]
+
+export const splitObject = <T extends UnknownObject>(
+  object: T
+): SplittedObjectArray<T> => {
+  if (Object.keys(object).length === 0) {
+    return []
+  }
+
+  return Object.entries(object).map(([key, value]) => ({
+    [key]: value
+  })) as SplittedObjectArray<T>
+}
```

Comment:
```suggestion
): SplittedObjectArray<T> => (Object.entries(object).map(([key, value]) => ({
    [key]: value
  })) as SplittedObjectArray<T>) || []
```

요렇게 하면 더 간결해질 것 같은데 어떠신가융??